hnn_core/network.py

```diff
@@ -301,7 +299,20 @@ def gid_to_type(self, gid):
             if gid in gids:
                 return gidtype
 
-    def _create_all_src(self):
+    def _get_src_type_and_pos(self, gid):
+        """Source type, position and whether it's a cell or artificial feed"""
+
+        # get type of cell and pos via gid
+        src_type = self.gid_to_type(gid)
+        type_pos_ind = gid - self.gid_dict[src_type][0]
+        src_pos = self.pos_dict[src_type][type_pos_ind]
+
+        real_cell_types = ['L2_pyramidal', 'L5_pyramidal',
+                           'L2_basket', 'L5_basket']
```

Comment:
this should probably be a global if it's used in other functions but I think we can leave it here for now.